yt/data_objects/tests/test_profiles.py

```diff
@@ -319,3 +325,62 @@ def test_profile_override_limits():
     assert_equal(ds.arr(obins, "g/cm**3"), profile.x_bins)
     assert_equal(ds.arr(obins, "dyne"), profile.y_bins)
     assert_equal(ds.arr(obins, "erg"), profile.z_bins)
+
+class TestBadProfiles(unittest.TestCase):
+
+    tmpdir = None
+    curdir = None
+
+    def setUp(self):
+        self.tmpdir = tempfile.mkdtemp()
+        self.curdir = os.getcwd()
+        os.chdir(self.tmpdir)
+
+    def tearDown(self):
+        os.chdir(self.curdir)
+        # clean up
+        shutil.rmtree(self.tmpdir)
+
+    def test_unequal_data_shape_profile(self):
```

Comment:
These tests need to be decorated with `@requires_module('h5py')` I *think* just decorating the test methods is sufficient but you can double-check locally by uninstalling h5py and running just these tests.